test/UportSubprovider.js

```diff
@@ -12,6 +12,10 @@ const address = '0x122bd1a75ae8c741f7e2ab0a28bd30b8dbb1a67e'
 const mnid = '2oeXufHGDpU51bfKBsZDdu7Je9weJ3r7sVG'
 const badMnid = '2nSX6hxNMgvgP9MtvoJDgSjVHGRsTuxpyPi'
 
+function encodeSignature({r,s,v}) {
+  return `${r.padStart(64, '0')}${s.padStart(64, '0')}${v}`
```

Comment:
@rmw2  I’m not 100% sure but I think it expects this to have a `0x` prefix. Can you look at the spec and make sure. If it’s good then merge it in, if not just add that.

Reply:
Yes it is definitely required from EIP712:

> DATA: Signature. As in eth_sign it is a hex encoded 129 byte array starting with 0x. It encodes the r, s and v parameters from appendix F of the yellow paper in big-endian format. Bytes 0…64 contain the r parameter, bytes 64…128 the s parameter and the last byte the v parameter. Note that the v parameter includes the chain id as specified in EIP-155.